addons/udes_priorities/models/stock_picking.py

```diff
@@ -0,0 +1,122 @@
+from odoo import fields, models, api, _
+from collections import defaultdict, OrderedDict
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.model
+    def _default_priority(self):
+        priorities = self.env["udes_priorities.priority"].search(self._priority_domain(), limit=1)
+        return priorities.reference if priorities else None
+
+    priority = fields.Selection(selection="get_priorities_for_selection", default=_default_priority)
+
+    def _priority_and_priority_group_domain(self):
+
+        domain = []
+        picking_type_ids = []
+
+        picking_types = self.mapped("picking_type_id")
+        default_picking_type_id = self.env.context.get("default_picking_type_id", None)
+
+        if picking_types:
+            picking_type_ids = picking_types.ids
+        elif default_picking_type_id:
+            picking_type_ids = [default_picking_type_id]
+
+        if picking_type_ids:
+            domain.extend(
+                [
+                    "|",
+                    ("picking_type_ids", "in", picking_type_ids),
+                    ("picking_type_ids", "=", False),  # No picking types means all picking types
+                ]
+            )
+
+        return domain
+
+    def _priority_domain(self):
+        # _priority_domain and _priority_group_domain  may diverge in certian circumstances
```

Comment:
typo: double space before `may`
typo: `certain`

same for `_priority_group_domain` below